x-pack/plugins/lens/public/types.ts

```diff
@@ -300,12 +300,16 @@ export type UserMessagesDisplayLocationId = UserMessageDisplayLocation['id'];
 export interface UserMessage {
   uniqueId: string;
   severity: 'error' | 'warning' | 'info';
+  hidePopoverIcon?: boolean;
   shortMessage: string;
   longMessage: string | React.ReactNode | ((closePopover: () => void) => React.ReactNode);
   fixableInEditor: boolean;
   displayLocations: UserMessageDisplayLocation[];
 }
 
+export type UseHandledMessage = Pick<UserMessage, 'hidePopoverIcon' | 'longMessage'> & {
```

Comment:
Rather than define a new type, we could define a new prop `hideIcon` on the `UserMessage` and reuse the same type there.
This will make the `customBadgeMessages` able to just return a list of messages to render, without the embeddable doing any re-mapping.
In the future, the new prop would be used also by us if we want to.